client/allocrunner/taskrunner/task_runner.go

```diff
@@ -291,6 +291,26 @@ func NewTaskRunner(logger *log.Logger, config *config.Config,
 		},
 	}
 
+	if tc.alloc.Job.ParentID != "" {
+		tc.baseLabels = append(tc.baseLabels, metrics.Label{
+			Name:  "parent_id",
+			Value: tc.alloc.Job.ParentID,
+		})
+		if strings.Contains(tc.alloc.Job.Name, "/dispatch-") {
+			tc.baseLabels = append(tc.baseLabels, metrics.Label{
+				Name:  "dispatch_id",
+				Value: strings.Split(tc.alloc.Job.Name, "/dispatch-")[1],
```

Comment:
See below comment. Consider pulling this into a helper function to reduce code duplication.

Reply:
thanks for the input, I wrote helper function hour ago and now I'm trying to figure where to place it =)

Reply:
In this case, its probably fine to not introduce a helper method. This and the other place we do this, are in two different packages and we would have to put one helper method in a util package that both can import. Doesn't seem worth it this time..

Reply:
That what's stopped me - 2 different packages without imports intersection.
Thanks @preetapan, and thanks @chelseakomlo for really usefull code review tips!